3a.train_individual_models/nbconverted/5.vis_probabilities.r

```diff
@@ -0,0 +1,159 @@
+  arrange(desc(median_healthy_proba))  # Sort in descending order of median
+
+# Step 2: Reorder the 'Metadata_treatment' factor levels in reverse order for top-to-bottom plotting
+filtered_df <- filtered_df %>%
+  mutate(Metadata_treatment = factor(Metadata_treatment, levels = rev(treatment_order$Metadata_treatment)))
+
+# Step 3: Create the ridge plot with reordered treatments
+height <- 15
+width <- 24
+options(repr.plot.width = width, repr.plot.height = height)
+
+# Create a ridge plot excluding DMSO and using all other treatments on the y-axis
+ridge_plot_non_DMSO <- ggplot(filtered_df, aes(x = predicted_probas, y = Metadata_treatment, fill = Metadata_Pathway)) +
+  geom_density_ridges(alpha = 0.7, scale = 2.25, rel_min_height = 0.01, bandwidth = 0.1) +  # Ridges colored by Pathway
+  geom_vline(xintercept = 1, linetype = "dashed", color = "black") +
+  scale_x_continuous(breaks = seq(0, 1, 0.5)) +
+  labs(x = "Probability of healthy prediction", y = "Treatment", color = "Pathway") +  # Update axis labels
+  scale_fill_manual(values = custom_palette) +  # Use the custom color palette
+  theme_bw() +
+  theme(
+    legend.position = "right",  # Position the legend on the right
+    legend.title = element_text("Pathway", size = 22),
+    axis.text = element_text(size = 22),
+    axis.text.x = element_text(size = 22),
+    axis.title = element_text(size = 24),
+    strip.text = element_text(size = 18),
+    strip.background = element_rect(
+      colour = "black",
+      fill = "#fdfff4"
+    ),
+    legend.text = element_text(size = 18),  # Adjust legend text size if needed
+    legend.key.size = unit(1.5, "lines")  # Adjust the size of the legend keys
+  ) +
+  facet_wrap(~ model_name, ncol = 5)  # Facet by model applied
+
+# Save figure
+ggsave(paste0(figures_dir, "/treatments_only_ridge_plot.png"), ridge_plot_non_DMSO, height = height, width = width, dpi = 500)
+
+ridge_plot_non_DMSO
+
+# Custom color palette inspired by Dark2 (includes one more color for a pathway missing in above plot)
+custom_palette <- c(
+  "#1b9e77",
+  "#d95f02",
+  "#7570b3",
+  "#e7298a",
+  "#66a61e",
+  "#984ea3",
+  "#e6ab02",
+  "#a6761d",
+  "#667665",
+  "#b3b3b3",
+  "#8dd3c7",
+  "#fb8072"
+)
+
+# Calculate the median predicted probability and cell count per compound per model
+summary_data <- filtered_df %>%
+    group_by(Metadata_treatment, model_name) %>%
+    summarise(
+        median_predicted_proba = median(predicted_probas),
+        cell_count = n(),
+        .groups = "drop"
+    ) %>%
+    left_join(
+        filtered_df %>% distinct(Metadata_treatment, Metadata_Pathway),
+        by = "Metadata_treatment"
+    )
+
+height <- 10
+width <- 14
+options(repr.plot.width = width, repr.plot.height = height)
+
+# Generate scatterplot for probability and cell count
+count_probas_plot <- ggplot(summary_data, aes(x = cell_count, y = median_predicted_proba, color = Metadata_Pathway, shape = model_name)) +
```

Comment:
THIS IS SUCH A COOL PLOT! 🎉 

Are you ready when Josh/Tim/Marion ask about which compounds are which? Alternatively, you could apply a `ggrepel` label - this could happen later as well, after merging this PR

Reply:
Thanks, I agree! Also, I think I will be ready! I agree we should use `ggrepel` to label the points, but that can be added after I fix the combined_batch1 training issue (data leakage).